### PR TITLE
OCPBUGS-12890: Add Ignition Stub for proxy data to GCP

### PR DESCRIPTION
**Pull request by barbacbd**
** bootstrap process was missing proxy data in the ignition file. Create a a proxy stub and merge the new ignition file with the original file in terraform.

### CODE REVIEW

**Comment by jianli-wei** [comment]
@barbacbd Please see my [comment](https://issues.redhat.com/browse/OCPBUGS-12890?focusedId=22580122&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22580122) in the bug, thanks!

File: data/data/gcp/bootstrap/main.tf [hunk 1]
```diff
@@ -29,6 +29,12 @@ resource "google_storage_bucket_object" "ignition" {
   content = var.ignition_bootstrap
 }
 
+resource "google_storage_bucket_object" "proxy" {
+  bucket  = google_storage_bucket.ignition.name
+  name    = "proxy.ign"
+  content = var.gcp_bootstrap_proxy_ignition_stub
+}
```
Comment (patrickdillon) [inline data/data/gcp/bootstrap/main.tf:36]: I don't think we need a bucket to hold the proxy ignition.

File: data/data/gcp/bootstrap/main.tf [hunk 2]
```diff
@@ -55,10 +61,20 @@ data "google_storage_object_signed_url" "ignition_url" {
   credentials = var.gcp_create_bootstrap_sa ? base64decode(google_service_account_key.bootstrap[0].private_key) : null
 }
 
+data "google_storage_object_signed_url" "proxy_url" {
+  bucket      = google_storage_bucket.ignition.name
+  path        = "proxy.ign"
+  duration    = "1h"
+  credentials = var.gcp_create_bootstrap_sa ? base64decode(google_service_account_key.bootstrap[0].private_key) : null
+}
+
```
Comment (patrickdillon) [inline data/data/gcp/bootstrap/main.tf:70]: This too, I don't think we need to create a signed url for the proxy ignition.

File: pkg/asset/ignition/bootstrap/bootstrap_ignition.go [hunk 3]
```diff
@@ -50,6 +50,14 @@ func GenerateIgnitionShimWithCertBundleAndProxy(bootstrapConfigURL string, userC
 		},
 	}
 
+	if bootstrapConfigURL != "" {
+		ign.Ignition.Config = igntypes.IgnitionConfig{
+			Replace: igntypes.Resource{
+				Source: ignutil.StrToPtr(bootstrapConfigURL),
```
Comment (patrickdillon) [inline pkg/asset/ignition/bootstrap/bootstrap_ignition.go:56]: You probably need to drop the previous assignment above
```
					Source: ignutil.StrToPtr(bootstrapConfigURL),
```

**Comment by patrickdillon** [comment]
/lgtm cancel 

I think this still needs some work.

**Pull request closed without merging**